### PR TITLE
LPS-190249 Adapt import/export

### PR DESCRIPTION
**Pull request by jkappler**
https://liferay.atlassian.net/browse/LPS-190249

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by sandrodw3** [comment]
❤️

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java [hunk 1]
```diff
@@ -130,11 +130,13 @@ private Object _toFormSuccessSubmissionResult(
...
 
-		if (successMessageJSONObject == null) {
+		if ((!saveInlineContent && !saveMappingConfiguration) ||
+			(successMessageJSONObject == null)) {
+
 			return null;
 		}
 
-		if (saveInlineContent && successMessageJSONObject.has("message")) {
+		if (successMessageJSONObject.has("message")) {
```
Comment (ruben-pulido) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java:141]: If we remove this, but saveMappingConfiguration is true, we will wrongly export message, won't we?

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java [hunk 2]
```diff
@@ -143,7 +145,7 @@ private Object _toFormSuccessSubmissionResult(
 			};
 		}
 
-		if (saveInlineContent && successMessageJSONObject.has("url")) {
+		if (successMessageJSONObject.has("url")) {
```
Comment (ruben-pulido) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java:176]: If we remove this, but saveMappingConfiguration is true, we will wrongly export url, won't we?

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java [hunk 3]
```diff
@@ -152,9 +154,7 @@ private Object _toFormSuccessSubmissionResult(
 			};
 		}
 
-		if (saveMappingConfiguration &&
-			successMessageJSONObject.has("layout")) {
-
+		if (successMessageJSONObject.has("layout")) {
```
Comment (ruben-pulido) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java:185]: If we remove this, but saveInlineContent is true, we will wrongly export layout, won't we?

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java [hunk 4]
```diff
@@ -130,20 +132,48 @@ private Object _toFormSuccessSubmissionResult(
...
 		}
 
-		if (saveInlineContent && successMessageJSONObject.has("url")) {
+		String type = successMessageJSONObject.getString("type");
+
+		if (Objects.equals(type, "none")) {
+			return new MessageFormSubmissionResult() {
+				{
+					messageType = MessageType.NONE;
+					showNotification = successMessageJSONObject.getBoolean(
```
Comment (ruben-pulido) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java:157]: Should we avoid exporting default value "false"?

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/importer/structure/util/FormLayoutStructureItemImporter.java [hunk 5]
```diff
@@ -252,8 +253,30 @@ private JSONObject _getSuccessMessageJSONObject(
...
+			if (Objects.equals(
+					formSuccessSubmissionResultMap.get("messageType"),
+					MessageFormSubmissionResult.MessageType.NONE.getValue())) {
+
+				return JSONUtil.put(
+					"notificationText",
+					_getLocalizedValuesJSONObject(
+						"message", formSuccessSubmissionResultMap)
+				).put(
+					"showNotification",
```
Comment (ruben-pulido) [inline modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/importer/structure/util/FormLayoutStructureItemImporter.java:265]: Should we avoid setting default value false if showNotification is not present in formSuccessSubmissionResultMap?

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/importer/structure/util/FormLayoutStructureItemImporter.java [hunk 6]
```diff
@@ -252,8 +253,30 @@ private JSONObject _getSuccessMessageJSONObject(
...
 
 		if (formSuccessSubmissionResultMap.containsKey("message")) {
-			return _getLocalizedValuesJSONObject(
-				"message", formSuccessSubmissionResultMap);
+			if (Objects.equals(
+					formSuccessSubmissionResultMap.get("messageType"),
+					MessageFormSubmissionResult.MessageType.NONE.getValue())) {
+
+				return JSONUtil.put(
+					"notificationText",
```
Comment (ruben-pulido) [inline modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/importer/structure/util/FormLayoutStructureItemImporter.java:261]: Should we avoid setting notificationText if _getLocalizedValuesJSONObject returns an empty json object?

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/importer/structure/util/FormLayoutStructureItemImporter.java [hunk 7]
```diff
@@ -252,8 +253,30 @@ private JSONObject _getSuccessMessageJSONObject(
...
+					GetterUtil.getBoolean(
+						formSuccessSubmissionResultMap.get("showNotification"))
+				).put(
+					"type", "none"
+				);
+			}
+
+			return JSONUtil.put(
+				"message",
+				_getLocalizedValuesJSONObject(
```
Comment (ruben-pulido) [inline modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/importer/structure/util/FormLayoutStructureItemImporter.java:275]: Should we avoid setting message if _getLocalizedValuesJSONObject returns an empty json object?

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java [hunk 8]
```diff
@@ -130,11 +130,13 @@ private Object _toFormSuccessSubmissionResult(
 		JSONObject successMessageJSONObject =
 			formStyledLayoutStructureItem.getSuccessMessageJSONObject();
 
-		if (successMessageJSONObject == null) {
+		if ((!saveInlineContent && !saveMappingConfiguration) ||
```
Comment (jkappler) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java:135]: @ruben-pulido, moved the check for saveInlineContent and saveMappingConfiguration to have an early return

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java (hunk 1 above)
Comment (jkappler) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java]: done, thanks ❤️

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java (hunk 2 above)
Comment (jkappler) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java]: done, thanks ❤️

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java (hunk 4 above)
Comment (jkappler) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/FormLayoutStructureItemMapper.java]: done, thanks ❤️

**Comment by jkappler** [comment]
done all the changes, thanks ❤️

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward:force

**Pull request closed without merging**